server/routes/apiRoute.js

```diff
@@ -74,54 +74,98 @@ app.get("/avatar/:name",(req,res)=>{
     res.sendFile(__dirname.split("\\").slice(0,-1).join("\\")+"/uploads/"+req.params.name)
 });
 
-// Zend een POST request dat de data uit de front-end in de database krijgt.
+// ---------------- REGISTRATIE ----------------
+
+// Zend een POST request dat de data uit de front-end in de database krijgt en daarmee een nieuwe gebruiker aanmaakt.
 app.post("/addgebruiker", upload.single('profielFoto'), async (req, res) => {
-    var data = req.body;
-    console.log(data.firstName);
+    let data = req.body;
+    let image = "defaultAvatar.png";
+
+    if (req.file !== undefined) {image = req.file.filename;}
     data.pass = await bcrypt.hash(data.pass, 10 );
-    console.log("Toevoeging gebruiker:");
-    connection.query("INSERT INTO gebruiker (firstName, lastName, email, pass, phone, birth, profielFotoLink, isWerkgever) VALUES (?,?,?,?,?,?,?,?)",[data.firstName, data.lastName, data.email, data.pass, data.phone, data.birth, req.file.filename,data.isWerkgever?1:0],
-        (error, results, fields) => {
-            if (error) {
-                console.log(error);
-                res.status(422).json;
-                res.json({message:error});
-            }else{
-                res.status(201).send("Gebruiker toegevoegd.");
-                console.log("Gebruiker toegevoegd.");
-
-                // Hier wordt het verificatie-email verstuurd. Wanneer we ook op andere plekken email gaan gebruiken kan deze code centraler opgeslagen worden.
-                const transporter = nodemailer.createTransport({
-                    service: 'gmail',
-                    auth: {
-                        user: 'roosteritHRO@gmail.com',
-                        pass: 'hogeschoolr'
-                    }
-                });
-
-                const mailOptions = {
-                    from: 'roosteritHRO@gmail.com',
-                    to: data.email,
-                    subject: 'Verificatie RoosterIt',
-                    html: ` 
-                    <h1>Geachte meneer/mevrouw ${data.lastName},</h1><p>Volg deze link om uw registratie te voltooien:</p>
-                    <p><a href='http://localhost:3000/emailverificatie/${data.email}'>Verifieer email</a></p>
-                    `
-                };
-
-                transporter.sendMail(mailOptions, function(error, info){
-                    if (error) {
-                        console.log(error);
-                    } else {
-                        console.log('Email verstuurd: ' + info.response);
-                    }
-                });
 
+    connection.query("INSERT INTO gebruiker (firstName, lastName, email, pass, phone, birth, profielFotoLink, isWerkgever) VALUES (?,?,?,?,?,?,?,?)",[data.firstName, data.lastName, data.email, data.pass, data.phone, data.birth,image ,data.isWerkgever?1:0],
+    (error, results, fields) => {
```

Comment:
Een gebruiker wordt als een werkgever toegevoegd in de db als hij geen werkgever is